src/core/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import {
+  Modal as MuiModal,
+  Button,
+  ModalProps as ModalBaseProps,
+} from '@material-ui/core';
+import React, { useState } from 'react';
+import useStyles from './styles';
+
+interface ModalProps extends ModalBaseProps {
+  hasCloseBtn: boolean;
+}
+
+const Modal: React.FC<ModalProps> = ({ children, hasCloseBtn }) => {
+  const classes = useStyles();
+  const [open, setIsOpen] = useState(true);
+
+  return (
+    <MuiModal
+      open={open}
+      onClose={() => setIsOpen(false)}
```

Comment:
If we want to add more features onClose we can use a `handleClose` and append that function here like

```suggestion
      onClose={() => {
           handleClose;
           setIsOpen(false);
      }}
```

Let's wait for other suggestion if we want this feature here or not.

Reply:
@arkajyotiMukherjee let me know your thoughts on this. Thanks

Reply:
I think that'll be done on the website it is used...onClose is fine

Reply:
I guess a simple onClose is fine by me